restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -281,6 +283,27 @@ public CallManager(final Configuration configuration, final ServletContext conte
         if (logger.isInfoEnabled()) {
             logger.info("CallManager extensions: "+(extensions != null ? extensions.size() : "0"));
         }
+        firstTimeCleanup();
+    }
+
+    private void firstTimeCleanup() {
+        if (logger.isInfoEnabled())
+            logger.info("Initial CallManager cleanup. Will check running state calls in DB and update state of the calls.");
+        String instanceId = RestcommConfiguration.getInstance().getMain().getInstanceId();
+        Sid sid = new Sid(instanceId);
+        final CallDetailRecordsDao callDetailRecordsDao = storage.getCallDetailRecordsDao();
+        List<CallDetailRecord> results= callDetailRecordsDao.getInCompleteCallDetailRecordsByInstanceId(sid);
+        for (CallDetailRecord result : results) {
```

Comment:
I have reduce the cost by the SQL statement. I don't have any idea on how to avoid this loop. Do you have any idea ?